### ISSUE TITLE
Text: Only allow DateTimeFormat enum instead of string value of it

### ISSUE THREAD

**Issue by nguhoa** [microsoft/Power-Fx#1760]
In Text, we currently allow both DateTimeFormat enum (LongDateTime) and string value of it ("'longdatetime'").

> Text( Now(), "ZZZ 'longdatetime' ZZZ" )
"ZZZ 'long14ateti8e' ZZZ"

> Text( Now(), "'longdatetime'" )
"Monday, August 14, 2023 10:32:19 AM"

> Text( Now(), "'longdatetime' ZZZ" )
"'long14ateti8e' ZZZ"

> Text( Now(), DateTimeFormat.LongDateTime )
"Monday, August 14, 2023 10:56:29 AM"

We will only support DateTimeFormat enum and not allow string alternative of it.

**Comment by gregli-msft** [comment]
Slightly more info...

Before we had strongly typed enums, enums and strings/numbers were interchangeable.  Which meant that the Format parameter to the Text function needed a way to support `DateTimeFormat.LongDate` as a string.  The solution was to have a special string, `'longdate'` in this case, that the Text function would key off of.  This is hacky and contradicts the normal handling of single ticks and other letters in the format string.  

For example, and see how the addition of another character changes the meaning completely:
```
1> Text( Now(), DateTimeFormat.LongDate )
"Wednesday, August 16, 2023"

2> Text( Now(), "'longdate'" )
"Wednesday, August 16, 2023"

3> Text( Now(), "'longdate'z" )
"'long16ate'z"
```

Times have changed and we now have strongly typed enums.  We should use it to differentiate from custom format strings.  As the [docs describe](https://learn.microsoft.com/en-us/power-platform/power-fx/reference/function-text), one should be able to use either the enum or a custom format string, but not both.  

In other words, example 2 above should look like example 3, without the extra `z`.

**Comment by nguhoa** [comment]
Fixed in https://github.com/microsoft/Power-Fx/pull/1761
